_internal/types.ts

```diff
@@ -259,7 +275,8 @@ export type Arguments =
   | undefined
   | false
 export type Key = Arguments | (() => Arguments)
-
+export type StrictTupleKey = ArgumentsTuple | null | undefined | false
+export type StrictKey = StrictTupleKey | (() => StrictTupleKey)
```

Comment:
Is that possible to replace the `Key` with this?

Reply:
Since we still need to support `useSWR('/api')` and `useSWR({ foo: 'bar' })`, we have to use different type here. 

Maybe we could deprecated `useSWR('/api')` and `useSWR({ foo: 'bar' })` in next major release ?